index.php

```diff
@@ -27,6 +27,46 @@
    exit;
 }
 
+//function to show warnings
+function warning($msg) { 
+	$script = "<script type='text/javascript'>$('#warning').fadeIn(1000);</script>";
+	$script .= "<script type='text/javascript'>$('#warning').html('<strong>Warning!</strong> $msg');</script>";
+	return $script;
+}
+
+//check if wol application (either wol or awake) is installed and set wol application to be used (e.g. for use with alpine & awake instead of wol)
+$wol_apps = [ "wakeonlan", "awake" ];//could be extended
+foreach ($wol_apps as $app) {
+  if (exec("command -v ".$app)<>"")
+  {
+    $wol = $app;//set the first application from the array that returns a non empty string as the wol application to be used
+    break;
+  }
+  if ($app === end($wol_apps))
+  {
+    $apps = implode(', ', $wol_apps);
+    $script = warning("unable to find any wol app, been looking for: <strong>".$apps."</strong> (one is necessary).<br>install one e.g. by running <strong>apt-get install wakeonlan</strong> or <strong>apk add awake</strong> (on the server)");//if non of the apps from the array are installed show a warning
+  }
+}
+
+//check if ping application is installed & working
+$ping_apps = [ "ping" ];//could be extended
+foreach ($ping_apps as $app) {
+  if (exec("command -v ".$app)<>"")
+  {
+    break;//just check it exists
+  }
+  if ($app === end($ping_apps))
+  {
+	$apps = implode(', ', $ping_apps);
+    $script = warning("Unable to find any ping app, been looking for: <strong>".$apps."</strong> (one is necessary).<br>Install one e.g. by running <strong>apt-get install iputils-ping</strong> or <strong>apk add iputils</strong> (on the server)");//if non of the apps from the array are installed show a warning
+  }
+}
+if (exec("ping -c 1 127.0.0.1 > /dev/null && echo $?")=="")
+{
+  $script = warning("The ping command is not working correctly. I tested this by pinging 127.0.0.1. This could be caused by a rights problem (are you root). An easy fix is usually to install iputils (apt-get install iputils-ping, apk add iputils");//if non of the apps from the array are installed show a warning
```

Comment:
My preferred approach to this issue is to give non-root users the ability to execute ping, which is what the new setup script ensures. In that case, the warning would change to: `Unable to execute ping on localhost. Ensure that the web user has ping permissions.`